### DISCUSSION TITLE
`no-this-alias` should apply only to `const`, but not to `let`

### DISCUSSION THREAD

**Discussion by GermanJablo** [typescript-eslint/typescript-eslint#7057] [Technical Discussions]
### RFC

This rule makes sense for `const`, but not for `let`.
It is normal for a variable to take the initial value of `this`, and then use it to iterate.

### Additional Info

Any additional info...

### Before you submit your RFC, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
Hi there - as per our contributing guidelines - we don't use discussions for small change requests. Please file an issue instead.

https://typescript-eslint.io/contributing/discussions
